src/soundbase.cpp

```diff
@@ -247,10 +258,51 @@ void CSoundBase::ParseCommandLineArgument ( const QString& strMIDISetup )
             iCtrlMIDIChannel = slMIDIParams[0].toUInt();
         }
 
+        bool bSimple = true;
+
         // [offset for level]
         if ( slMIDIParams.count() >= 2 )
         {
-            iMIDIOffsetFader = slMIDIParams[1].toUInt();
+            int i = slMIDIParams[1].toUInt (&bSimple);
+            if (bSimple)
+                iMIDIOffsetFader = i;
+        }
+
+        if (bSimple)
+        {
+            for (int i = 0; i < MAX_NUM_CHANNELS; i++)
+            {
+                if (i + iMIDIOffsetFader > 127)
+                    break;
+                aMidiCtls[i + iMIDIOffsetFader] = { EMidiCtlType::Fader, i };
+            }
+            return;
+        }
+
+        // We have named controllers
+
+        for (int i = 1; i < slMIDIParams.count(); i++)
+        {
+            QString sParm = slMIDIParams[i].toLower ().remove ( ' ' );
+            if ( sParm.isEmpty() )
+                continue;
+
+            int iCtrl = QString ( sMidiCtlChar ).indexOf ( sParm[0] );
+            if (iCtrl < 0)
+                continue;
+            EMidiCtlType eTyp = static_cast<EMidiCtlType>( iCtrl );
+
+            const QStringList slP = sParm.mid ( 1 ).split ( '*' );
+            int iFirst = slP[0].toUInt ();
+            int iNum = ( slP.count() > 1 ) ? slP[1].toUInt () : 1;
+            for (int iOff = 0; iOff < iNum; iOff++)
```

Comment:
You seem to have given up on the spacing here.  Could you just get these done, too please.

Reply:
I am sorry, but I simply am unable to figure out what the supposed spacing is.  It is not supported by any indentation engine and no other project appears to use it, and there is no description anywhere.  Could you point out just what lines need changing in what manner?

Reply:
As far as I know, you e.g. need spaces (not sure how many) in the for loop.

Example: 
`for(int` etc. should be `for ( int`. Just check the other code.

Reply:
Yep, it's definitely a pretty unique style and takes some getting used to.  Volker found me difficult to train, too... :)

There should always be matched spaces within paired delimiters -- so no only the `for (int i....)` needing to be `for ( int i... )` but also the `slP[1]` needing to be `slP[ 1 ]`.

Methods with argument lists have a space before and after the braces.  Methods without argument lists don't.  So `toUInt ()` should be `toUInt()`.

If you look back to last year when little was happening, you'll see Volker was making many of this sort of change when code was contributed.  Things are getting busy now, so it does really need to be either pushed back to the contributor (sorry) or we drop it (which will be messy).

Reply:
It's common in projects I work with to pick a standard (as in: used by a relevant number of people) formatting style.  If you pick one that is supported well by all of astyle, Emacs, and Clang, formatting the source once with the formatter people can best agree on (for reasonably standard styles there are usually only some detail differences) and then keep to it.  It's a nuisance for outstanding work/branches which need to get merged/rebased once.  And there will be small amounts of churn when intermediately the code gets run through the formatter to bring almost-but-not-quite contributions in line.

Typically some common style best matching the original style is picked, but to 

The Emacs style can even be prescribed in a directory-local variable in the root directory, but one can add rule files for the other formatters when the style is not exhaustively described by a name.  Bit of a distraction, but nowhere near a deterrent for new contributors than the current code (and obviously the current style does not cover, say, the include Opus libraries at all).

Anyway, I'll take a look whether I can manage to incorporate your hints.  I am constantly fighting my editor to get there when the normal situation would be that my editor helps.

Reply:
Just FYI:
```
origin:protocol.cpp:    for (int i = 0, j = 0; i < iDataLen; i++, j += 2 )
origin:recorder/jamrecorder.cpp:    for (int iChID = 0; iChID < vecptrJamClients.size(); iChID++)
origin:recorder/jamrecorder.cpp:    for (int i = 0; i < jamClientConnections.count(); i++ )

```

Reply:
> Yep, it's definitely a pretty unique style and takes some getting used to. Volker found me difficult to train, too... :)
> 
> There should always be matched spaces within paired delimiters -- so no only the `for (int i...)` needing to be `for ( int i... )` but also the `slP[1]` needing to be `slP[ 1 ]`.

I don't buy the latter.  Please do
~~~
git grep '\[' src
~~~
and look at the output.  `slP[ 1 ]` would be seriously out of sync with what the rest of the code does.  As I said: this is a mess where it is hard to figure out just what makes something particular require spaces and what not.  But with regard to index brackets, you appear to be wrong.